test/stop/residual_norm_kernels.cpp

```diff
@@ -110,6 +110,28 @@ TYPED_TEST(ResidualNorm, CanIgorneResidualNorm)
                  gko::NotSupported);
 }
 
+TYPED_TEST(ResidualNorm, CheckIfResZeroConverges)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using NormVector = typename TestFixture::NormVector;
+    auto initial_res = gko::initialize<Mtx>({10.0}, this->exec);
+    std::shared_ptr<gko::LinOp> rhs = gko::initialize<Mtx>({10.0}, this->exec);
+    auto criterion =
+        this->factory->generate(nullptr, rhs, nullptr, initial_res.get());
+    {
+        auto res_norm = gko::initialize<NormVector>({0.0}, this->exec);
+        constexpr gko::uint8 RelativeStoppingId{1};
+        bool one_changed{};
+        gko::array<gko::stopping_status> stop_status(this->ref, 1);
+        stop_status.get_data()[0].reset();
+        stop_status.set_executor(this->exec);
+
+        ASSERT_TRUE(criterion->update().residual_norm(res_norm).check(
+            RelativeStoppingId, true, &stop_status, &one_changed));
+        ASSERT_TRUE(stop_status.get_data()[0].has_converged());
+        ASSERT_TRUE(one_changed);
```

Comment:
```suggestion
            RelativeStoppingId, true, &stop_status, &one_changed));
        stop_status.set_executor(this->ref);
        ASSERT_TRUE(stop_status.get_data()[0].has_converged());
        ASSERT_TRUE(one_changed);
```
It should solve the segmentation fault

Reply:
I'd generally prefer if I could just setup the residual_norm to 0 and test that against the stopping status.
the whole step with the criterion seems to be unnecessary. Though I have no clue how to fix that if that is at all possible.

Reply:
Also should I check with rel_factory and abs_factory?